### PR TITLE
Refactor transformer conversion

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java [hunk 1]
```diff
@@ -25,238 +23,153 @@
...
+                angle0, r0, x0, g0, b0);
     }
 
-    /**
-     * Get the nominal magnetizing susceptance of a three windings transformer, located on bus star side.
-     */
-    public static double getB1(ThreeWindingsTransformer.Leg leg, boolean twtSplitShuntAdmittance) {
-        return getB1(leg, getCurrentPosition(leg.getRatioTapChanger()), getCurrentPosition(leg.getPhaseTapChanger()), twtSplitShuntAdmittance);
+    public static Integer getCurrentPosition(RatioTapChanger rtc) {
+        return rtc != null ? rtc.getTapPosition() : null;
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java:128]: I am not sure to understand why I have initialy chosen `null` if there is not ratio tap changer.

File: src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java [hunk 2]
```diff
@@ -25,238 +23,153 @@
...
     }
 
-    public static double getB1(ThreeWindingsTransformer.Leg leg, Integer rtcPosition, Integer ptcPosition, boolean twtSplitShuntAdmittance) {
-        return getB1(twtSplitShuntAdmittance ? leg.getB() / 2 : leg.getB(), leg.getRatioTapChanger(), leg.getPhaseTapChanger(), rtcPosition, ptcPosition);
+    public static Integer getCurrentPosition(PhaseTapChanger rtc) {
+        return rtc != null ? rtc.getTapPosition() : 0;
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java:132]: `ptc` should be better. I am not sure to understand why I have initialy chosen `0` when we have no phase tap changer.

File: src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java:128]: It was to not confuse with a possible value

File: src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java:132]: I renamed to ptc and replace 0 by null to be consistent with the other method.

**Review by annetill — APPROVED** [review]
No regression on functional tests.

**Pull request merged**